code/datums/diseases/advance/symptoms/necropolis.dm

```diff
@@ -91,18 +94,19 @@
 	if(!.)
 		return
 	var/mob/living/carbon/M = A.affected_mob
-	to_chat(M, span_danger("You feel weakened as the necropolis' blessing leaves your body."))
-	M.remove_movespeed_modifier(MOVESPEED_ID_NECRO_VIRUS_SLOWDOWN)
-	M.dna.species.punchdamagelow = initial(M.dna.species.punchdamagelow)
-	M.dna.species.punchdamagehigh = initial(M.dna.species.punchdamagehigh)
-	M.dna.species.punchstunthreshold = initial(M.dna.species.punchstunthreshold)
-	M.dna.species.brutemod /= 0.6
-	M.dna.species.burnmod /= 0.6
-	M.dna.species.heatmod /= 0.6
-	REMOVE_TRAIT(M, TRAIT_PIERCEIMMUNE, DISEASE_TRAIT)
-	if(fireproof)
-		REMOVE_TRAIT(M, TRAIT_RESISTHIGHPRESSURE, DISEASE_TRAIT)
-		REMOVE_TRAIT(M, TRAIT_RESISTHEAT, DISEASE_TRAIT)
-		M.weather_immunities -= "ash"
-		M.weather_immunities -= "lava"
+	if(fullpower)								//undo the changes if we got the buff
+		to_chat(M, span_danger("You feel weakened as the necropolis' blessing leaves your body."))
+		M.remove_movespeed_modifier(MOVESPEED_ID_NECRO_VIRUS_SLOWDOWN)
+		M.dna.species.punchdamagelow -= 4
+		M.dna.species.punchdamagehigh -= 4
+		M.dna.species.punchstunthreshold -= 1
+		M.dna.species.brutemod /= 0.6
```

Comment:
i think these would be better as setting them back to initial if nothing else is really modifying them, as division can get a little messy.

http://www.byond.com/docs/ref/#/proc/initial

Reply:
I thought we learned our lesson about this kind of thing from theos and I abusing shitcode to have near 0 brute/burn mods. No I'm not changing it, it's already getting multiplied by 0.6 as a prerequisite for this to even happen so it shouldn't be an issue. Even if it does do some wonky rounding that's at most like a 1% difference, practically nothing.